packages/e2e-test-utils/src/get-all-blocks.js

```diff
@@ -9,5 +9,29 @@ import { wpDataSelect } from './wp-data-select';
  * @return {Promise} Promise resolving with an array containing all blocks in the document.
  */
 export async function getAllBlocks() {
-	return wpDataSelect( 'core/block-editor', 'getBlocks' );
+	return (
+		( await wpDataSelect( 'core/block-editor', 'getBlocks' ) ) ||
+		page.evaluate( () => {
```

Comment:
@david-szabo97 the only change I made here is having the recursion as a fallback because in most cases it will not be needed. What do you think?